ratis-server-api/src/main/java/org/apache/ratis/server/protocol/RaftServerProtocol.java

```diff
@@ -38,4 +40,6 @@ enum Op {REQUEST_VOTE, APPEND_ENTRIES, INSTALL_SNAPSHOT}
   InstallSnapshotReplyProto installSnapshot(InstallSnapshotRequestProto request) throws IOException;
 
   StartLeaderElectionReplyProto startLeaderElection(StartLeaderElectionRequestProto request) throws IOException;
+
+  ReadIndexReplyProto readIndex(ReadIndexRequestProto request) throws IOException;
```

Comment:
Let's add only the async method. This sync'ed method is going to cause problem.  Let's don't add it.  We may skip Netty and SimulatedRPC for the moment, i.e. they won't support non-leader readIndex for now.